app/routes.py

```diff
@@ -1 +1,217 @@
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return make_response({"task":{
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": False
+        }}, 201)
+
+@task_bp.route("", methods=["GET"])
+def read_all_tasks():
+
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc":
+        task = Task.query.order_by(Task.title.asc())
+    elif sort_query == "desc":
+        task = Task.query.order_by(Task.title.desc())
+    else:
+        task = Task.query.all()
+
+    task_response = []
+    for task in task:
+        task_response.append(task.to_dict())
+        
+    return jsonify(task_response)
+
+@task_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+    return {"task": task.to_dict()}
+
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    return make_response({"task": task.to_dict()})
+ 
+def slack_bot_message(message):
+    PATH = "https://slack.com/api/chat.postMessage"
+    SLACK_API_KEY = os.environ.get("SLACK_API_KEY")
+
+    query_params = {
+        "channel" : "#task-notification",
+        "text" : message
+    }
+
+    requests.post(PATH, params=query_params,headers={"Authorization": SLACK_API_KEY})
+
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def complete_incomplete_task(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = datetime.utcnow()
+    
+    db.session.commit()
+
+    slack_bot_message(f"Someone just completed the task {task.title}")
+
+    return make_response({"task": task.to_dict()})
+
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_complete_task_incomplete(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = None
+
+    db.session.commit()
+
+    return make_response({"task": task.to_dict()})
+        
+
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response({"details":f'Task {task_id} "{task.title}" successfully deleted'})
+
+################## GOAL ROUTES ######################
+
+
+@goal_bp.route("", methods=["POST"])
+def handle_goals():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal(title=request_body["title"])
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return make_response({"goal":{
+            "id": new_goal.goal_id,
+            "title": new_goal.title
+        }}, 201)
```

Comment:
Consider putting this logic into a method like to_dict() in the Goal class (like you did for Task) so that you can reuse them method in your routes when you need to return a dict in the response